x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/mapper/Mapper.java

```diff
@@ -141,16 +145,23 @@ else if (aggregate.groupings()
             return new TopNExec(topN.source(), mappedChild, topN.order(), topN.limit(), null);
         }
 
-        if (unary instanceof Rerank rerank) {
-            mappedChild = addExchangeForFragment(rerank, mappedChild);
-            return new RerankExec(
-                rerank.source(),
-                mappedChild,
-                rerank.inferenceId(),
-                rerank.queryText(),
-                rerank.rerankFields(),
-                rerank.scoreAttribute()
-            );
+        // Inference operations must execute on coordinator, not on remote clusters
+        if (unary instanceof InferencePlan) {
+            mappedChild = addExchangeForFragment(unary.child(), mappedChild);
+            return switch (unary) {
+                case Rerank r -> new RerankExec(
+                    r.source(),
+                    mappedChild,
+                    r.inferenceId(),
+                    r.queryText(),
+                    r.rerankFields(),
+                    r.scoreAttribute()
+                );
+                case Completion c -> new CompletionExec(c.source(), mappedChild, c.inferenceId(), c.prompt(), c.targetField());
```

Comment:
Have we checked that there's no way to get to a place where we have an unexecutable plan?
IMO we should verify that there is a pipeline breaker before any of the inference plans.